### ISSUE TITLE
FATAL error if host unreachable

### ISSUE THREAD

**Issue by enrichz** [thomasmauerer/hassio-addons#40]
The addon crashes if the host where the network share resides is unreachable.This happens when I restart HA or the addon itself.
This is what i got from the logs:

```
[15:43:09] INFO: Host: 192.168.1.XXX
[15:43:09] INFO: Share: ha-backups
[15:43:09] INFO: Target Dir: prod
[15:43:09] INFO: Keep local: 5
[15:43:09] INFO: Keep remote: all
[15:43:09] INFO: Trigger time: 23:32
[15:43:09] INFO: Trigger days: Sun
[15:43:23] WARNING: PING 192.168.1.XXX (192.168.1.XXX): 56 data bytes

--- 192.168.1.XXX ping statistics ---
4 packets transmitted, 0 packets received, 100% packet loss
[15:43:23] FATAL: The provided host is unreachable. Please check your config.
[cmd] /run.sh exited 1
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
```

The problem is that my nas comes online just before the backup starts, it's not always on.
Is there a way to work around this? 
Thanks

**Comment by thomasmauerer** [comment]
Yes, you could set the `no_icmp` option to true. Then the add-on will skip the ping request at startup completely.
See here: https://github.com/thomasmauerer/hassio-addons/blob/master/samba-backup/DOCS.md#option-no_icmp

**Comment by enrichz** [comment]
I tried what you suggested and I get this:
```
[20:09:42] WARNING: do_connect: Connection to 192.168.1.XXX failed (Error NT_STATUS_HOST_UNREACHABLE)
[20:09:42] FATAL: Cannot access share. Unknown reason.
[cmd] /run.sh exited 1
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
```

**Comment by thomasmauerer** [comment]
Sorry I did not read carefully. Well, that's really a show-stopper at the moment if the NAS is not available during startup. I could make the entire pre-checks optional for such use-cases.

**Comment by enrichz** [comment]
That would be much appreciated, I don't know if someone else has the same use case as mine.
Hope you'll consider adding the functionality, the addon is really helpful.

**Comment by thomasmauerer** [comment]
Included in v4.4

**Comment by enrichz** [comment]
I can confirm that it's working, thanks.
